transformer_engine/common/fused_attn/fused_attn.cpp

```diff
@@ -91,7 +102,8 @@ NVTE_Fused_Attn_Backend nvte_get_fused_attn_backend(
         ((qkv_format == NVTE_QKV_Format::NVTE_BSHD) ||
          (qkv_format == NVTE_QKV_Format::NVTE_SBHD)) &&
         ((attn_mask_type == NVTE_Mask_Type::NVTE_CAUSAL_MASK) ||
-         (attn_mask_type == NVTE_Mask_Type::NVTE_NO_MASK))))) {
+         (attn_mask_type == NVTE_Mask_Type::NVTE_NO_MASK)))) &&
+      supported_ragged_offset_size) {
```

Comment:
This section is for FP8. We have two versions of FP8 implementation. The older implementation supports T3HD but int32 regardless of cuDNN version. The newer implementation doesn't support THD. So I think we can maybe do this here:
```&& !requires_64bit_ragged_offset```